### PR TITLE
Bug 1362520 - Add telemetry for plugin notifications to main summary dataset

### PR DESCRIPTION
**Pull request by wlach**

### CODE REVIEW

File: src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala [hunk 1]
```diff
@@ -485,7 +487,15 @@ object MainSummaryView {
         asInt(simpleMeasures \ "main"),
         asInt(simpleMeasures \ "firstPaint"),
         asInt(simpleMeasures \ "sessionRestored"),
-        asInt(simpleMeasures \ "totalTime")
+        asInt(simpleMeasures \ "totalTime"),
+
+        // bug 1362520 - plugin notifications
+        asInt(histograms \ "PLUGINS_NOTIFICATION_SHOWN" \ "sum"),
```
Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: you can use the `hsum` function for these.

File: src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala [hunk 2]
```diff
@@ -592,6 +602,12 @@ object MainSummaryView {
     }
   }
 
+  def buildPluginNotificationUserActionSchema = StructType(List(
+    StructField("allow_now", IntegerType),
```
Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: Please add `nullable = true` for these fields

File: src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala [hunk 3]
```diff
@@ -586,6 +588,104 @@ class MainSummaryViewTest extends FlatSpec with Matchers{
...
+      "values":{"1":1,"0":0,"2":0},
+      "bucket_count":3,
+      "sum":1
+    }
+  }"""),
+      None);
+
+    val header = Header(message.toByteArray.length)
+
+    val framedMessage = {
```
Comment (mreid-moz) [inline src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala]: I don't think we want to repeat this logic here - let's either add some helpers to `moztelemetry` to make a framed message or just call `messageToRow` on `message` and check that the resulting summary row looks right (more like the `"Heka records" can "be summarized"` test below).

**Comment by wlach** [comment]
@mreid-moz updated based on your review

File: src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala [hunk 4]
```diff
@@ -586,6 +588,74 @@ class MainSummaryViewTest extends FlatSpec with Matchers{
     }
   }
 
+  "MainSummary plugin counts" can "be serialized" in {
+    val sparkConf = new SparkConf().setAppName("MainSummary")
```
Comment (mreid-moz) [inline src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala]: I don't think you need the Spark context anymore (nor the `ByteArray*` imports above).

Comment (wlach) [inline src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala]: Argh, sorry I missed taking those out.

**Comment by wlach** [comment]
@mreid-moz are we good to land after rebasing?

File: src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala [hunk 5]
```diff
@@ -586,6 +588,69 @@ class MainSummaryViewTest extends FlatSpec with Matchers{
     }
   }
 
+  "MainSummary plugin counts" can "be serialized" in {
```
Comment (mreid-moz) [inline src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala]: one last nit: s/erial/ummar/

**Pull request merged**